newrelic-agent/src/main/java/com/newrelic/agent/config/UtilizationDataConfig.java

```diff
@@ -35,7 +35,7 @@
 
     private String parseConfigVal(String key) {
         if (getProperty(key) != null) {
-            return String.valueOf(getProperty(key));
+            return getProperty(key).toString();
```

Comment:
On Java 8 Agent, String.valueOf was causing a ClassCastException with a cross agent unit test.   

```
java.lang.ClassCastException: 
class java.lang.Long cannot be cast to class 
[C (java.lang.Long and 
[C are in module java.base of loader 'bootstrap')
```

When debugging, the more readable reason is :

```
Type mismatch Can't assign non-array value to an array
```

On Java 7 Agent, the byte code for this method is

```
String.valueOf:(Ljava/lang/Object;)Ljava/lang/String;
```

On Java 8, the byte code is:

```
String.valueOf:([C)Ljava/lang/String;
```

I don't know why this compiles differently. I tried a scratch program with just a psvm and the same behavior.  

This took way too long to track down. In the interest of time, this was the quickest fix and I'm not sure if it is the best one.  

This method `parseConfigValue` is only called in this class so the blast radius appears small.